.github/workflows/build-cppfront.yaml

```diff
@@ -0,0 +1,40 @@
+name: Multi-platform Build of cppfront
+on: push
+jobs:
+  build-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ilammy/msvc-dev-cmd@v1
+      - name: Compiler name & version
+        run: cl.exe
+      - name: Build
+        run: cl.exe source/cppfront.cpp -std:c++latest -MD -EHsc -experimental:module -W4
+  build-unix-like:
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-latest]
+        compiler: [g++-10, g++-11, g++-12, clang++-12, clang++-14]
+        cxx-std: ['c++20', 'c++2b']
+        exclude:
+          # GCC 10 doesn't have support for c++23
+          - compiler: g++-10
+            cxx-std: 'c++2b'
+        include:
+          - runs-on: macos-latest
+            compiler: clang++
+            cxx-std: 'c++20'
+    runs-on: ${{ matrix.runs-on }}
+    env:
+      CXX: ${{ matrix.compiler }}
+      CXXFLAGS: -std=${{ matrix.cxx-std }} -Wall -Wextra -Wold-style-cast -pthread
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install compiler
+      if: startsWith(matrix.runs-on, 'ubuntu')
+      run: sudo apt-get install -y ${{ matrix.compiler }}
```

Comment:
Should this be `$CXX` instead of `${{ matrix.compiler }}` for consistency, or can that not be used there?

Reply:
I think it could be referenced directly yeah, but defining the `CXX ` envvar is also super common (specially when using build systems that expect those vars to be set).

Reply:
> I think it could be referenced directly yeah

I was actually saying the opposite:

    run: sudo apt-get install -y $CXX

Reply:
Done!